astropy/utils/tests/test_data.py

```diff
@@ -2165,7 +2165,16 @@ def test_download_ftp_file_properly_handles_socket_error():
     with pytest.raises(urllib.error.URLError) as excinfo:
         download_file(faulty_url)
     errmsg = excinfo.exconly()
-    assert "Name or service not known" in errmsg or 'getaddrinfo failed' in errmsg or 'Temporary failure in name resolution' in errmsg  # noqa
+    found_msg = False
+    possible_msgs = ['Name or service not known',
+                     'nodename nor servname provided, or not known',
+                     'getaddrinfo failed',
+                     'Temporary failure in name resolution']
+    for cur_msg in possible_msgs:
+        if cur_msg in errmsg:
+            found_msg = True
+            break
+    assert found_msg, f'Got {errmsg}, expected one of these: {",".join(possible_msgs)}'
```

Comment:
```suggestion
    found_msg = [cur_msg for cur_msg in possible_msgs if cur_msg in errmsg]
    assert len(found_msg) > 0, f'Got {errmsg}, expected one of these: {",".join(possible_msgs)}'
```
Experimented with a slightly more concise version, but that's entirely a matter of taste. :)

Reply:
Hmm. I am not sure because while more concise, the suggested logic would loop through everything regardless. Maybe I should keep this open for a bit to see if anyone else has a strong preference one way or another.